### ISSUE TITLE
improvement of `RULES.md`

### ISSUE THREAD

**Issue by nbehrnd** [markdownlint/markdownlint#437]
## Describe the Enhancement:
If I copy-paste `/markdownlint/docs/RULES.md` from the forked project e.g., to the desktop, and subsequently check this very file by

```shell
mdl ./RULES.md
```

there are multiple instances where the linter reports an issue.  Of course, some of them need to be questionable examples to highlight (in contrast) how to write *better* markdown.  Yet, not all of the 100 instances (`mdl RULES.md | wc -l`) belong to this type.  E.g., a some dozen of examples yield `MD046` because the code block is not fenced.

+ Is file `RULES.md` generated automatically? 
+ If file `RULES.md` *is not* generated/updated automatically, how could I fence an example of markdown in a .md file to manually correct some of the entries?

   So far, I know that three back ticks and e.g., key `python` as leading, and a line with three back ticks as closing fence works for Python

```shell
\```python
print("This is python")
\```
```

(Sorry, I don't know how to escape back ticks in markdown used to fence a snippet of code.)

   So far, no example for markdown was seen.  But would be/is the corresponding key for markdown `md`, or `markdown`?

### Impacted Rules:
No rules are impacted.

## Describe the Need:
The advantage would be that the linter's own documentation would follow/would follow closer than now the rules the linter checks.  The observations refer to mdl (version 0.12.0) in an instance of Linux Debian 12/bookworm (branch testing).
